src/app/util/ember-compatibility-functions.cpp

```diff
@@ -331,8 +361,9 @@ CHIP_ERROR ReadViaAccessInterface(FabricIndex aAccessingFabricIndex, bool aIsFab
 {
     AttributeValueEncoder::AttributeEncodeState state =
         (aEncoderState == nullptr ? AttributeValueEncoder::AttributeEncodeState() : *aEncoderState);
-    AttributeValueEncoder valueEncoder(aAttributeReports, aAccessingFabricIndex, aPath, kTemporaryDataVersion, aIsFabricFiltered,
-                                       state);
+    DataVersion version = kTemporaryDataVersion;
+    ReadClusterDataVersion(aPath.mEndpointId, aPath.mClusterId, version);
```

Comment:
Why is this call not error checked? And why do we need a notion of a temporary data version?

Reply:
we should do the error check, we should remove temporary data version, and use kUndefinedDataVersion, thanks